docs/reference/search/knn-search.asciidoc

```diff
@@ -121,10 +121,14 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn-k]
 include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn-num-candidates]
 
 `query_vector`::
-(Required, array of floats)
+(Optional, array of floats)
 include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn-query-vector]
 ====
 
+`query_vector_builder`::
+(Optional, object)
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn-query-vector-builder]
+
```

Comment:
This API is deprecated. We shouldn't really add features to it. Maybe it occurs by proxy by adding it to the `knn` query.

Reply:
on top of modifying `docs/reference/query-dsl/knn-query.asciidoc`, should I just revert this change, then?

Reply:
I think so @tteofili. this API is deprecated and folks shouldn't be using it. Everything is under `_search` now. Either in the `knn` clause or the `knn` query.